frontend/src/components/EditableLabelsDescriptionListGroup.tsx

```diff
@@ -1,223 +1,170 @@
-  const [isAddLabelModalOpen, setIsAddLabelModalOpen] = React.useState(false);
-  const [addLabelInputValue, setAddLabelInputValue] = React.useState('');
-  const addLabelInputRef = React.useRef<HTMLInputElement>(null);
-  let addLabelValidationError: string | null = null;
-  if (reservedKeys.includes(addLabelInputValue)) {
-    addLabelValidationError = 'Label must not match an existing label or property key';
-  } else if (addLabelInputValue.length > 63) {
-    addLabelValidationError = "Label text can't exceed 63 characters";
-  }
 
-  const toggleAddLabelModal = () => {
-    setAddLabelInputValue('');
-    setIsAddLabelModalOpen(!isAddLabelModalOpen);
-  };
-  React.useEffect(() => {
-    if (isAddLabelModalOpen && addLabelInputRef.current) {
-      addLabelInputRef.current.focus();
-    }
-  }, [isAddLabelModalOpen]);
-
-  const addLabelModalSubmitDisabled = !addLabelInputValue || !!addLabelValidationError;
-  const submitAddLabelModal = (event?: React.FormEvent) => {
-    event?.preventDefault();
-    if (!addLabelModalSubmitDisabled) {
-      addUnsavedLabel(addLabelInputValue);
-      toggleAddLabelModal();
+  const handleEditComplete = (_event: MouseEvent | KeyboardEvent, newText: string) => {
+    const error = validateLabel(newText);
+    if (error) {
+      setLabelErrors((prev) => ({ ...prev, [newText]: error }));
+    } else {
+      setUnsavedLabels((prev) => [...prev, newText]);
+      setLabelErrors({});
     }
   };
 
   return (
-    <>
-      <DashboardDescriptionListGroup
-        title={title}
-        isEmpty={labels.length === 0}
-        contentWhenEmpty={contentWhenEmpty}
-        isEditable={!isArchive}
-        isEditing={isEditing}
-        isSavingEdits={isSavingEdits}
-        contentWhenEditing={
+    <DashboardDescriptionListGroup
+      data-testid="editable-labels-group"
+      title={title}
+      isEmpty={labels.length === 0}
+      contentWhenEmpty={contentWhenEmpty}
+      isEditable={!isArchive}
+      isEditing={isEditing}
+      isSavingEdits={isSavingEdits}
+      isSaveDisabled={Object.keys(labelErrors).length > 0}
+      contentWhenEditing={
+        <>
           <LabelGroup
-            data-testid="label-group"
+            data-testid="editable-label-group"
             isEditable={!isSavingEdits}
-            numLabels={unsavedLabels.length}
+            numLabels={10}
+            expandedText="Show Less"
+            collapsedText="Show More"
             addLabelControl={
               !isSavingEdits && (
                 <Label
-                  textMaxWidth="40ch"
+                  data-testid="add-label-button"
                   color="blue"
                   variant="outline"
-                  isOverflowLabel
-                  onClick={toggleAddLabelModal}
+                  isEditable
+                  editableProps={{
+                    'aria-label': 'Add label',
+                    defaultValue: '',
+                    'data-testid': 'add-label-input',
+                  }}
```

Comment:
The way I understand the mockup, I don't think we want the "Add label" button to be an editable label itself. It can just be a button like before, but instead of its `onClick` opening a modal, it should add a new label called "New label" via `setUnsavedLabels()`.

Reply:
+1 to @mturley.
@YuliaKrimerman here's a PF demo of how should the `Add label` work: https://www.patternfly.org/components/label/#editable-label-group-with-add-button Hope it helps.